### PR TITLE
Refactor documentation related to container development and SSH connectivity

### PR DESCRIPTION
**Pull request by karolz-ms**
This is the change that we agreed on during last doc review call on Friday. I have pulled the content related to recommended use of Remote-SSH extension into "Choosing your dev environment" topic and left the advanced SSH tunneling option in a separate topic, not shown in the TOC.

@bwateratmsft FYI

### CODE REVIEW

**Comment by karolz-ms** [comment]
@ghogen @gregvanl please review--thx!!

File: docs/containers/choosing-dev-environment.md [hunk 1]
```diff
@@ -72,13 +72,27 @@ Windows Subsystem for Linux represents a great choice for container-based servic
...
+After Docker is installed and working on the remote machine, you can use VS Code's [Remote - SSH](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) extension from the [Remote Development](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) extension pack to connect to your remote machine and work there.
 
-**Reusing the host Docker daemon**
+1. Run command **Remote-SSH: Add new SSH host...** and follow the prompts to set up a connection to the target host.
+
+1. Run command **Remote-SSH: Connect to host...** and connect to the host.
+
+1. A new VS Code window opens, running in the context of the target machine. If using password authentication, the password will be prompted here. It is recommended to set up [SSH key authentication](https://www.ssh.com/ssh/public-key-authentication), for ease of use.
```
Comment (ghogen) [inline docs/containers/choosing-dev-environment.md]:
```suggestion
1. A new VS Code window opens, running in the context of the target machine. If you're using password authentication, the password will be prompted here. We recommend that you set up [SSH key authentication](https://www.ssh.com/ssh/public-key-authentication), for ease of use.
```

Comment (ghogen) [reply docs/containers/choosing-dev-environment.md]: Making it a little friendlier / easier to read.

File: docs/containers/choosing-dev-environment.md [hunk 2]
```diff
@@ -72,13 +72,27 @@ Windows Subsystem for Linux represents a great choice for container-based servic
...
+
+1. In the Extensions view, install the Docker extension (on the remote host) (a reload may be required after this step):
+
+   ![Screenshot - Installing the Docker extension](images/devenv/install-in-ssh.png)
+
+>**NOTE**: If you are using the Docker extension to build Docker images and have source code, the approach above probably means you have your source enlistment on the remote host, rather than on your developer workstation. If you are just using the Docker extension for the Docker Explorer features, then you can disregard this.
+
+### Local VM
+
+To use a virtual machine running on your developer workstation, you need to enable **nested virtualization** option in your virtualization software. Nested virtualization is supported by all mainstream virtualization technologies such as Hyper-V, Parallels, or Oracle VirtualBox. Then you can [install Docker](https://docs.docker.com/install/) in the same way as you would install it on a remote machine.
```
Comment (ghogen) [inline docs/containers/choosing-dev-environment.md]: Would it be appropriate to add a link to the Hyper-V virtualization docs: https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization

File: docs/containers/choosing-dev-environment.md [hunk 3]
```diff
@@ -72,13 +72,27 @@ Windows Subsystem for Linux represents a great choice for container-based servic
...
 
 Alternatively, you can install just the Docker CLI inside development environment and point the CLI to the Docker host (daemon) running on the developer workstation using [Docker context mechanism](https://docs.docker.com/engine/context/working-with-contexts/). The main concern with this approach is to ensure network connectivity from the VM to the Docker daemon on the host, and to do so in a secure way. One option is to use [SSH tunneling](/docs/containers/ssh.md) to developer workstation. Another option is to [make Docker daemon listen on HTTPS port](https://docs.docker.com/engine/security/https/). Both options are considered advanced and outside the scope of this document.
```
Comment (ghogen) [inline docs/containers/choosing-dev-environment.md]:
```suggestion
Alternatively, you can install just the Docker CLI inside your development environment and point the CLI to the Docker host (daemon) running on the developer workstation using [Docker context mechanism](https://docs.docker.com/engine/context/working-with-contexts/). The main concern with this approach is to ensure network connectivity from the VM to the Docker daemon on the host, and to do so in a secure way. One option is to use [SSH tunneling](/docs/containers/ssh.md) to the developer workstation. Another option is to [make Docker daemon listen on HTTPS port](https://docs.docker.com/engine/security/https/). Both options are considered advanced and outside the scope of this document.
```

Comment (ghogen) [reply docs/containers/choosing-dev-environment.md]: Consider deleting the "outside the scope of this document" part -- you've already provided a link to the instructions, so no need to apologize or explain why it's not covered here.

File: docs/containers/ssh.md [hunk 4]
```diff
@@ -1,36 +1,13 @@
...
-1. In the Extensions view, install the Docker extension (on the remote host) (a reload may be required after this step):
-
-   ![Screenshot - Installing the Docker extension](images/ssh/install-in-ssh.png)
-
->**NOTE**: If you are using the Docker extension to build Docker images and have source code, the approach above probably means you have your source enlistment on the remote host, rather than your local machine. If you are just using the Docker extension for the Docker Explorer features, then you can disregard this.
-
-## Directly via SSH
-
-It is possible to connect to a remote Docker daemon over SSH without using VS Code remoting, but it is more complicated. This is only recommended if you cannot have your source code on the Docker daemon server.
+It is possible to connect to a remote Docker engine over SSH without [using VS Code remoting extension](/docs/containers/choosing-dev-environment.md#remote-machine), but it is an advanced scenario that requires good SSH configuration skills:
```
Comment (ghogen) [inline docs/containers/ssh.md]: I'd be cautious about the part that says it requires "good SSH configuration skills" as some users may react negatively, thinking that the documentation's job is to explain this well enough that anyone could do it.

File: docs/containers/choosing-dev-environment.md (hunk 1 above)
Comment (bwateratmsft) [reply docs/containers/choosing-dev-environment.md]: I kinda think we should put "strongly" in there. Key auth is far more secure.

File: docs/containers/ssh.md (hunk 4 above)
Comment (bwateratmsft) [reply docs/containers/ssh.md]: Agree on that one.

File: docs/containers/choosing-dev-environment.md (hunk 1 above)
Comment (karolz-ms) [reply docs/containers/choosing-dev-environment.md]: @bwateratmsft agreed, will do

File: docs/containers/choosing-dev-environment.md (hunk 2 above)
Comment (karolz-ms) [reply docs/containers/choosing-dev-environment.md]: Will do

File: docs/containers/ssh.md (hunk 4 above)
Comment (karolz-ms) [reply docs/containers/ssh.md]: I have changed this topic intro to say:

We recommend to use [VS Code Remote-SSH extension](/docs/containers/choosing-dev-environment.md#remote-machine) to connect to a remote machine running Docker engine, but it also possible to connect to the remote Docker engine directly, using SSH tunneling:

**Comment by karolz-ms** [comment]
@ghogen thank you for all the suggestions. I believe I have addressed all of them.

**Pull request merged**